internal/store/elasticsearch/discovery_search_repository.go

```diff
@@ -33,6 +33,10 @@ func (repo *DiscoveryRepository) Search(ctx context.Context, cfg asset.SearchCon
 	if maxResults <= 0 {
 		maxResults = defaultMaxResults
 	}
+	offset := cfg.Offset
```

Comment:
Should we enforce a limit of 10000 here for offset and max results? According to the Elasticsearch documentation:

> Note that `from + size` can not be more than the `index.max_result_window` index setting, which defaults to 10,000.

Reply:
Do you mean returning an error back to user ?

Reply:
Yes, basically a validation error.

Reply:
do we need to keep the limit 10k? 🤔  since the user can always change the `index.max_result_window` value at es cluster level.

Reply:
Can we take it as a configuration for the service in that case with 10k as the default? If you think this should be done in a separate task, that is okay with me as well.

Reply:
I'm assuming you mean ability to configure `index.max_result_window` from compass itself. If that's the case, lets take it separately. Have added a comment in code for documentation on this field.

Reply:
No, configuring Elasticsearch is left up to the user. We just add a field in our Config for `MaxSearchResults` that would default to `10000`.

Reply:
sure. Lets take it separate.